### PR TITLE
Update API and IMPL to MyFaces 4.0.1

### LINKED ISSUE #25354: Update faces-4.0 to MyFaces 4.0.1
**Describe the bug**  

The following issues are fixed in the 4.0.1 update: 

[[MYFACES-4469](https://issues.apache.org/jira/browse/MYFACES-4469)] - "java.lang.IllegalArgumentException: object is not an instance of declaring class" occurs in FacesConfigurator.purgeConfiguration
    [[MYFACES-4586](https://issues.apache.org/jira/browse/MYFACES-4586)] - [perf] Resource bundle not cached
    [[MYFACES-4588](https://issues.apache.org/jira/browse/MYFACES-4588)] - faces.getClientWindow does not retrieve client id
    [[MYFACES-4590](https://issues.apache.org/jira/browse/MYFACES-4590)] - NullPointerException when PrimeFaces FileUpload used in composite component
    [[MYFACES-4591](https://issues.apache.org/jira/browse/MYFACES-4591)] - [perf] remove duplicated CDI ELResolver
    [[MYFACES-4593](https://issues.apache.org/jira/browse/MYFACES-4593)] - Wrong Error recovery strategy - See also https://github.com/jakartaee/faces/issues/1800
    [[MYFACES-4594](https://issues.apache.org/jira/browse/MYFACES-4594)] - Web Context Parameter Logging -- Multi Threading Problems
    [[MYFACES-4596](https://issues.apache.org/jira/browse/MYFACES-4596)] - faces.ts style tags must retain their original position (and script tags as well)
[[MYFACES-4599](https://issues.apache.org/jira/browse/MYFACES-4599)] - Remove javadoc reference to org.apache.myfaces.SERIALIZE_STATE_IN_SESSION
[[MYFACES-4584](https://issues.apache.org/jira/browse/MYFACES-4584)] - faces.js /ts improvements post 4.0

Note that some issues have already been addressed in earlier versions of Liberty, such as MYFACES-4594. 

**Steps to Reproduce**  
Please see the JIRA issues for additional details.

**Expected behavior**  
Please see the JIRA issues for additional details.

**Diagnostic information:**  
 - OpenLiberty Version: 23.0.0.3-23.0.0.6
 - Affected feature(s) faces-4.0
 - Java Version: n/a
 - server.xml configuration: n/a

**Additional context**  
Release Notes: https://issues.apache.org/jira/secure/ReleaseNote.jspa?projectId=10600&version=12352978

### PR DESCRIPTION
**Pull request by volosied**
fixes #25354

### CODE REVIEW

**Comment by volosied** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.jsf.2.3_fat,com.ibm.ws.jsf.2.2_fat,io.openliberty.org.apache.myfaces.4.0_fat,com.ibm.ws.jsf_fat_lWAS_faces4.0

File: dev/io.openliberty.faces.4.0.thirdparty/bnd.bnd [hunk 1]
```diff
@@ -22,7 +22,7 @@ Fragment-Host: io.openliberty.org.apache.myfaces.4.0;bundle-version=1.0
 
 Bundle-Name: Open Liberty Jakarta MyFaces Third Party API
 Bundle-SymbolicName: io.openliberty.faces.4.0.thirdparty
-Implementation-Version: 4.0.0
+Implementation-Version: 4.0.1
```
Comment (pnicolucci) [inline dev/io.openliberty.faces.4.0.thirdparty/bnd.bnd:25]: copyright update

File: dev/io.openliberty.jakarta.faces.4.0/bnd.bnd [hunk 2]
```diff
@@ -30,8 +30,7 @@ DynamicImport-Package: \
   org.apache.myfaces.spi
 
 -includeresource: \
-	@${repo;org.apache.myfaces.core:myfaces-api;${Implementation-Version};EXACT}!/!(META-INF/maven/*|module-info.class),\
-  META-INF=@src/META-INF
+	@${repo;org.apache.myfaces.core:myfaces-api;${Implementation-Version};EXACT}!/!(META-INF/maven/*|module-info.class)
```
Comment (pnicolucci) [inline dev/io.openliberty.jakarta.faces.4.0/bnd.bnd:33]: I think all that changed here is spaces -> tab? Can you clean that up?

**Comment by pnicolucci** [comment]
@volosied  please make sure the JSF/Faces buckets run in full mode in your personal build.

File: dev/io.openliberty.jakarta.faces.4.0/bnd.bnd (hunk 2 above)
Comment (volosied) [reply dev/io.openliberty.jakarta.faces.4.0/bnd.bnd:33]: for this change, I removed the src/META-INF overlay, so the line before had the `,\` at the end removed

**Comment by volosied** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.jsf.2.3_fat,com.ibm.ws.jsf.2.2_fat,io.openliberty.org.apache.myfaces.4.0_fat,com.ibm.ws.jsf_fat_lWAS_faces4.0

**Comment by volosied** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.jsf.2.3_fat,com.ibm.ws.jsf.2.2_fat,io.openliberty.org.apache.myfaces.4.0_fat,com.ibm.ws.jsf_fat_lWAS_faces4.0

**Review by pnicolucci — DISMISSED** [review]

**Comment by volosied** [comment]
#libby

**Comment by volosied** [comment]
#libby

**Comment by volosied** [comment]
#libby

**Review by isaacrivriv — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**